Mathlib/Algebra/GroupWithZero/WithZero.lean

```diff
@@ -35,6 +35,9 @@ instance one : One (WithZero α) where
 
 @[simp, norm_cast] lemma coe_one : ((1 : α) : WithZero α) = 1 := rfl
 
+-- We give this higher priority so that unzero_coe doesn't trigger.
+@[simp high] lemma unzero_one (hx : (1 : WithZero α) ≠ 0) : unzero hx = 1 := rfl
```

Comment:
Could you add the `ofNat` version too?

Reply:
Ok, I added another version that adds an `ofNat`, I don't usually use this function though, so tell me if you wanted something else.

Reply:
I'm not seeing it, I'm afraid

Reply:
See the other conversation.